src/NexusMods.App.UI/Pages/MyGames/MyGamesViewModel.cs

```diff
@@ -22,98 +23,148 @@
+                                    await Task.Run(async () => await ManageGame(installation));
+                                    vm.State = GameWidgetState.ManagedGame;
+                                }
+                            );
+
+                            vm.RemoveAllLoadoutsCommand = ReactiveCommand.CreateFromTask(async () =>
+                                {
+                                    if (GetJobRunningForGameInstallation(installation).IsT2) return;
+
+                                    vm.State = GameWidgetState.RemovingGame;
+                                    await Task.Run(async () => await RemoveAllLoadouts(installation));
+                                    vm.State = GameWidgetState.DetectedGame;
+                                }
+                            );
+
+                            vm.ViewGameCommand = ReactiveCommand.Create(() => { NavigateToFirstLoadout(conn, installation); });
+
+                            vm.IsManagedObservable = Loadout.ObserveAll(conn)
+                                .Filter(l => l.IsVisible() && l.InstallationInstance.GameMetadataId == installation.GameMetadataId)
+                                .Count()
+                                .Select(c => c > 0);
+
+                            var job = GetJobRunningForGameInstallation(installation);
+
+                            // fixes when the page loads and a job is still running
+                            vm.State = job.Value switch
+                            {
+                                CreateLoadoutJob _ => GameWidgetState.AddingGame,
+                                UnmanageGameJob _ => GameWidgetState.RemovingGame,
+                                _ => GameWidgetState.DetectedGame
+                            };
+
+                            return vm;
+                        }
+                    )
+                    .Bind(out _installedGames)
+                    .SubscribeWithErrorLogging()
+                    .DisposeWith(d);
+
+                // NOTE(insomnious): The weird cast is so that we don't get a circular reference with Abstractions.Games when
+                // referencing Abstractions.GameLocators directly 
+                var supportedGamesAsIGame = gameRegistry.SupportedGames.Cast<IGame>();
+
+                var miniGameWidgetViewModels = supportedGamesAsIGame
+                    .Select(game =>
+                        {
+                            var vm = provider.GetRequiredService<IMiniGameWidgetViewModel>();
+                            vm.Game = game;
+                            vm.Name = game.Name;
+                            // is this supported game installed?
+                            vm.IsFound = _installedGames.Any(installation => installation.Installation.GetGame().Equals(game));
+                            vm.GameInstallations = _installedGames
+                                .Where(installation => installation.Installation.GetGame().Equals(game))
```

Comment:
I don't like this `Equals` check here, that is going to be a `object.Equals` check.

Reply:
I think a similar comparison was done using an ID. Let me see.